server/strategies/localSignIn.js

```diff
@@ -0,0 +1,27 @@
+import UserModel from '../models/UserModel'
+import { verifyHash } from '../utils/crypto/hash'
+
+export default async function (req, username, password, done) {
+  try {
+    const { appDb } = req.app.locals
+    const userAttributes = { uid: username }
+    const user = await UserModel.findOne(appDb, userAttributes, {
+      password: 1,
+      role: 1,
+      display_name: 1,
+      mail: 1,
+      icon: 1,
+      access: 1,
+      account_expires: 1,
+      uid: 1,
+    })
+    if (!verifyHash(password, user?.password))
+      return res.status(400).json({ error: 'Incorrect password' })
```

Comment:
should there be a different message if the user does not exist?

Reply:
Yes and it that depends on wether we are signing in or signing up. This signIn verifies the hash then the user is serialized and a session is created. 

When signing up in the AuthController.create method we have a condition to handle existing users.
```javascript 
    if (existingUser)
            return res.status(400).json({ error: 'User has an account' })
```